ComponentKit/TransactionalDataSources/Common/CKTransactionalComponentDataSourceState.mm

```diff
@@ -88,4 +89,15 @@ - (NSUInteger)hash
   return CKIntegerArrayHash(hashes, CK_ARRAY_COUNT(hashes));
 }
 
+static NSArray *flattenedModelsFromSections(NSArray *sections)
+{
+  NSMutableArray *models = [NSMutableArray new];
+  for (NSArray *section in sections) {
+    for (CKTransactionalComponentDataSourceItem *item in section) {
+      [models addObject:item.model];
+    }
+  }
+  return models;
+}
+
```

Comment:
If you have a case of

Section 1: Item 1
Section 2: Item 2, Item 3

and 

Section 1: Item 1, Item 2
Section 2: Item 3

Then this would consider them equal. Are you sure you want that behavior?